src/client/routes/Portfolio/index.tsx

```diff
@@ -24,124 +28,154 @@ const HeaderCard = styled(SummaryCard)`
-    // .plus(labsSummary.totalDeposits)
-    .toString();
-
-  const summaryCardItems = [
-    { header: t('dashboard.total-net-worth'), Component: <Amount value={netWorth} input="usdc" /> },
-  ];
-  if (walletIsConnected) {
-    summaryCardItems.push({
-      header: t('dashboard.available-deposit'),
-      Component: <Amount value={walletSummary.totalBalance} input="usdc" />,
-    });
-  }
-  if (currentNetworkSettings.earningsEnabled) {
-    summaryCardItems.push(
-      {
-        header: t('dashboard.vaults-earnings'),
-        Component: <Amount value={vaultsSummary.totalEarnings} input="usdc" />,
-      },
-      {
-        header: t('dashboard.vaults-est-yearly-yield'),
-        Component: <Amount value={vaultsSummary.estYearlyYeild} input="usdc" />,
+  const userTokensLoading = generalLoading && !userTokens.length;
+  const [currentRole, setRole] = useState<string>(BORROWER_POSITION_ROLE);
+  const [data, setdata] = useState<any[]>([]);
+  const [aggregatedCreditLinePage, setAggregatedCreditLine] = useState<CreditLinePage>();
+
+  const availableRoles = [BORROWER_POSITION_ROLE, LENDER_POSITION_ROLE];
+
+  const SummaryCardItems = availableRoles.map((role: string) => {
+    return {
+      header: t(''),
+      Component: (
+        <RoleOption onClick={() => handleSetRole(role)} active={role === currentRole} key={`s-${role}`}>
+          {t(`settings:${role}`)}
+        </RoleOption>
+      ),
+    };
+  });
+
+  const handleSetRole = (role: string) => {
+    setdata([]);
+    setRole(role);
+  };
+
+  useEffect(() => {
+    if (!userAddress || !isValidAddress(userAddress)) {
+      dispatch(AlertsActions.openAlert({ message: 'INVALID_ADDRESS', type: 'error' }));
+      return;
+    } else if (userAddress.length === 42) {
+      dispatch(LinesActions.getUserPortfolio({ user: userAddress.toLocaleLowerCase() }));
+    }
+  }, [currentRole, walletIsConnected]);
+
+  useEffect(() => {
+    if (userPortfolio && currentRole === BORROWER_POSITION_ROLE) {
+      //Types for returned obj need to be set up correctly
+      //@ts-ignore
+      const borrowerData: any[] = userPortfolio.borrowerLineOfCredits;
+      setdata(borrowerData);
+      if (borrowerData && borrowerData[0]) {
+        const lineId = borrowerData[0].id;
+        dispatch(LinesActions.setSelectedLineAddress({ lineAddress: lineId }));
+      }
+    }
+    if (userPortfolio && currentRole === LENDER_POSITION_ROLE) {
+      const lenderData = userPortfolio.lenderPositions.positions!;
+      console.log('lender positions', lenderData);
+      setdata(lenderData);
+      if (lenderData && lenderData[0].id) {
+        const lineId = lenderData[0].id;
+        dispatch(LinesActions.setSelectedLineAddress({ lineAddress: lineId }));
       }
-    );
-  }
+    }
+  }, [userPortfolio, currentRole]);
+
+  useEffect(() => {
+    let aggregate;
+    if (data && currentRole === BORROWER_POSITION_ROLE) {
+      aggregate = formatGetBorrowerQuery(data, userAddress);
```

Comment:
formatting should be done during querying process. should not be giving dirty data back to devs to handle

Reply:
I agree. My intention was for the data in `state.user.portfolio.borrowerLineOfCredits` to be used directly in the Portfolio component.

@Seroxdesign is there a particular reason you are using this function instead of the data stored in `state.user.portfolio.borrowerLineOfCredits`? It appears that you are only using the `positions` data within the object returned by `formatGetBorrowerQuery`.

Let me know if you'd like to hop on a quick call to discuss.

Reply:
I am using more than the positions data, we need to aggregate escrow and col events + other stuff we can jump on a call at some point

Reply:
Renamed to lenderData and now set Borrower data directly